ouroboros-network/src/Ouroboros/Network/Protocol/Trans/Hello/Type.hs

```diff
@@ -0,0 +1,130 @@
+{-# LANGUAGE DataKinds             #-}
+{-# LANGUAGE EmptyCase             #-}
+{-# LANGUAGE FlexibleInstances     #-}
+{-# LANGUAGE GADTs                 #-}
+{-# LANGUAGE InstanceSigs          #-}
+{-# LANGUAGE KindSignatures        #-}
+{-# LANGUAGE PolyKinds             #-}
+{-# LANGUAGE QuantifiedConstraints #-}
+{-# LANGUAGE ScopedTypeVariables   #-}
+{-# LANGUAGE TypeFamilies          #-}
+{-# LANGUAGE UndecidableInstances  #-}
+
+module Ouroboros.Network.Protocol.Trans.Hello.Type where
+
+import           Data.Void (Void)
+
+import           Network.TypedProtocol.Core
+import           Ouroboros.Network.Util.ShowProxy
+
+
+-- | 'Hello' protocol transformer.  It allows to transform a protocol
+-- which initial state is 'stIdle' for which the agancy might be held by
+-- the server, to a protocol with initial state 'StIdle' in which the client has
+-- agancy.  It adds 'MsgHello' which is a transition from 'StIdle' to
+-- @'StWrapped' stIdle@.
+--
```

Comment:
The initial state does not matter. It's polymorphic in the initial state. It only matters about the agency.

Isn't this more accurate as documentation:
```
-- | The 'Hello' protocol transformer reverses the initial agency of the protocol,
-- from one in which the server has the initial agency to one in which the client
-- has the initial agency.
--
-- It extends the underlying protocol with an initial extra 'MsgHello' message,
-- which reverses the agency between the two peers.
```

Reply:
Yes It does not matter in the sense its polymorphic, but for a reader it might be useful that the `MsgHello` moves the state to `stIdle`.    This is however visible in the types, I'll go with your wording.

Reply:
I already did that :)